src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java

```diff
@@ -90,6 +92,9 @@
   private static final int DEFAULT_SEND_AND_DEFER_BUFFER_SIZE = KB.toBytes(10);
   private static final String DEFAULT_DECOMPRESS_PROPERTY_NAME = SYSTEM_PROPERTY_PREFIX + "http.client.decompress";
 
+  private static final String DISABLE_REQUEST_STREAMING_PROPERTY_NAME = SYSTEM_PROPERTY_PREFIX + "http.client.decompress";
+  private static boolean requestStreamingDisabled = (System.getProperty(DISABLE_REQUEST_STREAMING_PROPERTY_NAME) != null);
```

Comment:
I think it might be best to follow the decompress property way but reading with `true` as default.